src/components/TokenExpired.js

```diff
@@ -1,25 +1,91 @@
 import React from 'react'
 import PropTypes from 'prop-types'
 import { connect } from 'react-redux'
+import { SessionTimeout } from 'patternfly-react'
 
 import ErrorAlert from './ErrorAlert'
 import { msg } from '_/intl'
 
+import { logout } from '_/actions'
+
+const MAX_INACTIVE_TIME = 60 * 20 // 20 minutes
+const TIME_TO_DISPLAY_MODAL = 30 // 30 seconds
+
 // TODO: allow the user to cancel the automatic reload?
 // If so, change config.isTokenExpired to false and add additional check to doCheckTokenExpired() before actual reload
-const TokenExpired = ({ config }) => {
-  if (!config.get('isTokenExpired')) {
-    return null
+class TokenExpired extends React.Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      expired: false,
+      counter: MAX_INACTIVE_TIME,
+    }
+
+    this.dropCounter = this.dropCounter.bind(this)
+    this.decrementCounter = this.decrementCounter.bind(this)
+
+    document.body.addEventListener('mousemove', this.dropCounter)
+    this.timer = setInterval(this.decrementCounter, 1000)
   }
 
-  return <ErrorAlert message={msg.authorizationExpired()} id='token-expired' />
+  dropCounter () {
+    if (!this.state.expired) {
+      this.setState({ counter: MAX_INACTIVE_TIME })
+    }
+  }
+  decrementCounter () {
+    const state = {
+      counter: this.state.counter - 1,
+    }
+    if (this.state.counter <= TIME_TO_DISPLAY_MODAL && !this.state.expired) {
+      state.expired = true
+    }
+    if (this.state.counter <= 0) {
+      this.props.onLogout()
+    }
+    this.setState(state)
+  }
+
+  componentWillUnmount () {
+    clearInterval(this.timer)
+  }
+
+  render () {
+    const { config, onLogout } = this.props
+    if (this.state.expired) {
```

Comment:
Nit-pick: So technically the inactivity timer has expired and now the "you have 30 seconds before auto-logout" timeout modal is displayed.  They have `MAX_INACTIVE_TIME` plus `TIME_TO_DISPLAY_MODAL`. :wink:

Reply:
No, modal is displayed, when only `TIME_TO_DISPLAY_MODAL` seconds left. So `MAX_INACTIVE_TIME` is time to user logout. `TIME_TO_DISPLAY_MODAL` is time when modal must be shown.